src/extension.ts

```diff
@@ -1,176 +1,208 @@
-'use strict';
-
 import * as vscode from 'vscode';
 import { fromNullable, Option } from 'fp-ts/lib/Option';
 import { getRandomInt } from './utils';
+import { EXTENSION_NAME, MATERIAL_LIST, LAST_THEME_MATERIAL, Messages, CommandsIds, SettingsKeys, LAST_SWITCH_DAY, ThemeTypes } from './enums';
 
-const LAST_THEME_MATERIAL = 'last-theme-is-material';
-const MATERIAL_LIST = [
-  'Material Theme',
-  'Material Theme High Contrast',
-  'Material Theme Darker',
-  'Material Theme Darker High Contrast',
-  'Material Theme Palenight',
-  'Material Theme Palenight High Contrast',
-  'Material Theme Ocean',
-  'Material Theme Ocean High Contrast',
-  'Material Theme Lighter',
-  'Material Theme Lighter High Contrast'
-];
-
-type ThemeObject = { id: string; label: string; [key: string]: string };
-type SaveThemeMessage = 'copyall' | 'removedtheme' | 'addedtheme';
-
-function getSaveMessages(
-  messageType: SaveThemeMessage,
-  opts: { number?: number; theme?: string } = { number: 0, theme: '' }
-): string {
-  switch (messageType) {
-    case 'copyall':
-      return `Copied ${opts.number} themes to settings`;
-    case 'addedtheme':
-      return `Added ${opts.theme} to Random Theme List in settings`;
-    case 'removedtheme':
-      return `Removed ${opts.theme} from Random Theme List in settings`;
-  }
-}
-
-function getThemeName(theme: ThemeObject): string {
-  return theme.id || theme.label;
-}
 
 function getUserSettings(): vscode.WorkspaceConfiguration {
   return vscode.workspace.getConfiguration();
 }
 
 function getExtensionConfig(): vscode.WorkspaceConfiguration {
-  return vscode.workspace.getConfiguration('randomThemeSwitcher');
+  return vscode.workspace.getConfiguration(EXTENSION_NAME);
 }
 
-async function changeTheme({
-  extensionConfig = getExtensionConfig(),
-  userSettings = getUserSettings(),
-  context
-}: {
-  context?: vscode.ExtensionContext;
-  extensionConfig?: vscode.WorkspaceConfiguration;
-  userSettings?: vscode.WorkspaceConfiguration;
-} = {}) {
-  const themeList = getThemeList(extensionConfig, userSettings);
+
+async function changeTheme(
```

Comment:
inserisci il tipo di ritorno della funzione, servirà in futuro per dei refactor

Reply:
Intendi Promise<void> ?

``` Typescript
async function changeTheme(
  context?: vscode.ExtensionContext): Promise<void> {
```